mobius/src/geometry/arc_angles.rs

```diff
@@ -0,0 +1,265 @@
+use std::{
+    error::Error,
+    f64::consts::{PI, TAU},
+    fmt::Display,
+};
+
+use crate::{float_error::FloatError, nearly::is_nearly};
+
+#[derive(Debug)]
+pub enum ArcAnglesParseError {
+    /// One of the input floats is not valid
+    BadFloat(FloatError),
+    /// Arc angles must be specified in either strictly increasing
+    /// or strictly decreasing order
+    NotStrictlyMonotone(f64, f64, f64),
+    /// Arc is bigger than a full circle, I'm not supporting this.
+    BigArcNotSupported(f64, f64, f64),
+}
+
+impl Display for ArcAnglesParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::BadFloat(err) => err.fmt(f),
+            Self::NotStrictlyMonotone(a, b, c) => write!(
+                f,
+                "angles must be either strictly increasing or strictly decreasing: ({}, {}, {})",
+                a, b, c
+            ),
+            Self::BigArcNotSupported(a, b, c) => write!(
+                f,
+                "only arcs smaller than a full circle are supported: ({}, {}, {})",
+                a, b, c
+            ),
+        }
+    }
+}
+
+impl From<FloatError> for ArcAnglesParseError {
+    fn from(value: FloatError) -> Self {
+        Self::BadFloat(value)
+    }
+}
+
+impl Error for ArcAnglesParseError {}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum ArcDirection {
+    // Counter-clockwise
+    Counterclockwise,
+    Clockwise,
+}
+
+/// Angles for use with CircularArc. These are subject to the following
+/// restrictions:
+///
+/// - a < b < c (CCW arc) or a > b > c (CW arc). This reduces corner cases
+/// - |c - a| < 2pi so we're always drawing less than a full circle
+/// - the value of a will be reduced to be in [0, 2pi)
+#[derive(Clone, Copy, Debug)]
+pub struct ArcAngles(pub f64, pub f64, pub f64);
```

Comment:
I'm starting to wonder - maybe instead of `(angle_a, angle_b, angle_c)`, maybe this should actually store `(angle_a, angle_b, direction)`? I need to think about this for a bit.

Reply:
if so... would I still have the 3-angle constructor? might be useful for computations in `ClineArc::classify()`, but for human construction, a constructor that just takes 2 angles might be nicer. 

E.g. if a < b, then the arc is clockwise
if a > b, then the arc is ccw
if a == b, then the arc is invalid
if abs(a - b) > 2 pi, then the arc is invalid (bigger than full circle).

So maybe 2 constructors?

```
from_angles(a, b, c) -> computes (reduced_a, reduced_c, CW/CCW)
new(a, b) -> computes(a, b, CW/CCW)
```

Reply:
Also, due to the reduced angles and sign convention for angles:

- You don't actually need to store the direction, it can be computed on the fly as it does now
- Since the value of b doesn't actually matter aside from computing ccw/cw, it could also be computed on the fly after construction by doing (a + b) / 2.

Reply:
Implemented this. feels a lot cleaner!